internal/extensionapi/utils.go

```diff
@@ -42,8 +43,16 @@ func GetNamespaceFromPath(path string) (string, error) {
 	return "", fmt.Errorf("cannot find the namespace in URL")
 }
 
-// GetUserFromHeaders extracts the user from request headers
+// GetUserFromHeaders extracts the user from Kubernetes request context
 func GetUserFromHeaders(r *http.Request) string {
+	// Try to get user info from Kubernetes request context
+	if userInfo, ok := request.UserFrom(r.Context()); ok {
+		if userInfo != nil && userInfo.GetName() != "" {
+			return stringutil.SanitizeUsername(userInfo.GetName())
+		}
+	}
+
+	// Fallback to headers for backward compatibility
```

Comment:
do you need backward compatibility here?
- if no, remove please
- if we want to keep the logic, remove the comment that justify it as backward-compatibility please